fabric-data-attachment-api-v1/src/main/java/net/fabricmc/fabric/api/attachment/v1/AttachmentRegistry.java

```diff
@@ -0,0 +1,149 @@
+
+import java.util.Objects;
+import java.util.function.Supplier;
+
+import com.mojang.serialization.Codec;
+import org.jetbrains.annotations.ApiStatus;
+
+import net.minecraft.util.Identifier;
+
+import net.fabricmc.fabric.impl.attachment.AttachmentRegistryImpl;
+
+/**
+ * Class used to create and register {@link AttachmentType}s. To quickly create {@link AttachmentType}s, use one of the various
+ * {@code createXXX} methods:
+ * <ul>
+ *     <li>{@link #create(Identifier)}: attachments will be neither persistent nor auto-initialized.</li>
+ *     <li>{@link #createDefaulted(Identifier, Supplier)}: attachments will be auto-initialized, but not persistent.</li>
+ *     <li>{@link #createPersistent(Identifier, Codec)}: attachments will be persistent, but not auto-initialized.</li>
+ * </ul>
+ *
+ * <p>For finer control over the attachment type and its properties, use {@link AttachmentRegistry#builder()} to
+ * get a {@link Builder} instance.</p>
+ */
+@ApiStatus.Experimental
+public final class AttachmentRegistry {
+	private AttachmentRegistry() {
+	}
+
+	/**
+	 * Creates <i>and registers</i> an attachment. The data will not be persisted.
+	 *
+	 * @param id  the identifier of this attachment
+	 * @param <A> the type of attached data
+	 * @return the registered {@link AttachmentType} instance
+	 */
+	public static <A> AttachmentType<A> create(Identifier id) {
+		Objects.requireNonNull(id, "identifier cannot be null");
+
+		return AttachmentRegistry.<A>builder().buildAndRegister(id);
+	}
+
+	/**
+	 * Creates <i>and registers</i> an attachment, that will be automatically initialized with a default value
+	 * when an attachment does not exist on a given target, using {@link AttachmentTarget#getAttachedOrCreate(AttachmentType)}.
+	 *
+	 * @param id          the identifier of this attachment
+	 * @param initializer the initializer used to provide a default value
+	 * @param <A>         the type of attached data
+	 * @return the registered {@link AttachmentType} instance
+	 */
+	public static <A> AttachmentType<A> createDefaulted(Identifier id, Supplier<A> initializer) {
+		Objects.requireNonNull(id, "identifier cannot be null");
+		Objects.requireNonNull(initializer, "initializer cannot be null");
+
+		return AttachmentRegistry.<A>builder()
+				.initializer(initializer)
+				.buildAndRegister(id);
+	}
+
+	/**
+	 * Creates <i>and registers</i> an attachment, that will persist across server restarts.
+	 *
+	 * @param id    the identifier of this attachment
+	 * @param codec the codec used for serialization
+	 * @param <A>   the type of attached data
+	 * @return the registered {@link AttachmentType} instance
+	 */
+	public static <A> AttachmentType<A> createPersistent(Identifier id, Codec<A> codec) {
+		Objects.requireNonNull(id, "identifier cannot be null");
+		Objects.requireNonNull(codec, "codec cannot be null");
+
+		return AttachmentRegistry.<A>builder().persistent(true).codec(codec).buildAndRegister(id);
```

Comment:
Hm, maybe then `Builder::persistent(Boolean)` should be replaced with `Builder::persistent(Codec)`? Is there any point in asking for boolean? Moreover, [it is impossible `Builder::build`](https://github.com/FabricMC/fabric/blob/19d21da1fe3cbc091a3122ce347978d2e81ff73c/fabric-data-attachment-api-v1/src/main/java/net/fabricmc/fabric/impl/attachment/AttachmentRegistryImpl.java#L86-L88) when the type is persistent but don't have a codec. Maybe even `Builder::codec` should be renamed to "serializable" or something

Reply:
Originally, an independent codec argument was added because we thought it would be useful for syncing. Syncing has been removed, but having a codec on hand might still be useful in the future. That being said, I agree the current way is a bit of boilerplate, I'll ask about what can be done.